### DISCUSSION TITLE
Documentation of `size` parameter

### DISCUSSION THREAD

**Discussion by mjuopperi** [python-pillow/Pillow#6849] [Ideas]
Various methods of the library take a `size` tuple with a doc string like "size – Requested size."

I assumed it would be `(width, height)`, but finding it from documentation was needlesly complex. Only reference I found was 
 [here](https://pillow.readthedocs.io/en/latest/handbook/concepts.html#size). 

Would it make sense to either improve documentation by changing the method docs to be clearer (e.g. "size – Requested size. (width, height)." or perhaps by using a NamedTuple or similar self documenting type?

**Comment by radarhere** [comment]
When you say "various methods", I only found [`draft()`](https://pillow.readthedocs.io/en/latest/reference/Image.html#PIL.Image.Image.draft) and [`thumbnail()`](https://pillow.readthedocs.io/en/latest/reference/Image.html#PIL.Image.Image.thumbnail). Are there more that I'm not seeing, or is that it?

**Reply by radarhere** [reply]
Thanks to #6851, `draft()` and `thumbnail()` have now been updated.

**Comment by mjuopperi** [comment]
- https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.resize
- https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.transform
etc.

[`Image.new`](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.new) has proper documentation: "size – A 2-tuple, containing (width, height) in pixels."

**Reply by hugovk** [reply]
> * [pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.resize](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.resize)

This says:

> size – The requested size in pixels, as a 2-tuple: (width, height).

That looks okay?

---

> * [pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.transform](https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.transform)

This says:

> size – The output size.

So that can be improved.

---

> * etc.

More examples welcome!

**Reply by radarhere** [reply]
https://pillow.readthedocs.io/en/stable/reference/Image.html#PIL.Image.Image.transform has now been updated.
